### ISSUE TITLE
Error when specifying user/pass and proxy

### ISSUE THREAD

**Issue by cwurm** [logstash-plugins/logstash-output-elasticsearch#516]
Supplying both `user` /  `password` and `proxy` causes an error message in 5.0.

Example config:
```
input {}
filter {}
output {
  elasticsearch {
    hosts => ["192.168.1.1:9200"]
    user => "elastic"
    password => "changeme"
    proxy => "http://random:8080"
  }
}
```

Error:
```
Sending Logstash's logs to /Users/cwurm/Products/Logstash/logstash-5.0.1/logs which is now configured via log4j2.properties
[2016-11-30T09:41:27,902][INFO ][logstash.outputs.elasticsearch] Elasticsearch pool URLs updated {:changes=>{:removed=>[], :added=>["http://~hidden~:~hidden~@192.168.1.1:9200"]}}
[2016-11-30T09:41:27,907][INFO ][logstash.outputs.elasticsearch] Using mapping template from {:path=>nil}
[2016-11-30T09:41:28,029][WARN ][logstash.outputs.elasticsearch] UNEXPECTED POOL ERROR {:e=>#<TypeError: can't convert Symbol into Integer>}
[2016-11-30T09:41:28,031][ERROR][logstash.outputs.elasticsearch] Failed to install template. {:message=>"can't convert Symbol into Integer", :class=>"TypeError"}
[2016-11-30T09:41:28,031][INFO ][logstash.outputs.elasticsearch] New Elasticsearch output {:class=>"LogStash::Outputs::ElasticSearch", :hosts=>["192.168.1.1:9200"]}
[2016-11-30T09:41:28,033][INFO ][logstash.pipeline        ] Starting pipeline {"id"=>"main", "pipeline.workers"=>4, "pipeline.batch.size"=>125, "pipeline.batch.delay"=>5, "pipeline.max_inflight"=>500}
[2016-11-30T09:41:28,035][INFO ][logstash.pipeline        ] Pipeline main started
[2016-11-30T09:41:28,123][INFO ][logstash.agent           ] Successfully started Logstash API endpoint {:port=>9600}
[2016-11-30T09:41:31,058][WARN ][logstash.agent           ] stopping pipeline {:id=>"main"}
```

It's not possible to process any data.

Seems to be in the similar part of the software as https://github.com/logstash-plugins/logstash-output-elasticsearch/issues/441

**Comment by andrewvc** [comment]
This looks like a bug in our HTTP client. I've opened up a PR to fix that here: https://github.com/cheald/manticore/pull/58

**Comment by andrewvc** [comment]
The first part of fixing this will be to abolish user/pass proxy declaration and only support URLs per #450 . The second part is then internally translating that canonical representation to *always* be a hash for Manticore. I'll open up a new PR there once #450 is merged. That PR should be very simple and fast to merge.
